app/src/app/components/Toolbar/ColorPicker.tsx

```diff
@@ -115,21 +133,43 @@ export const ColorPicker = <T extends boolean>({
   }
 
   return (
-    <div>
+    <div onMouseLeave={() => setColorSelectIndex(-1)}>
       <RadioGroupRoot
-        onValueChange={value => {
-          const index = colorArray.indexOf(value);
-          if (index !== -1) onValueChange(index, value);
+        onValueChange={(value: String) => {
+          const index = Number(value);
+          onValueChange(index, colorScheme[index]);
         }}
-        value={value !== undefined ? colorArray[value] : undefined}
-        defaultValue={colorArray[defaultValue]}
+        value={String(value)}
       >
         <Flex direction="row" wrap="wrap">
           {!!mapDocument &&
-            colorArray.slice(0, numDistricts).map((color, i) => (
+            colorScheme.slice(0, numDistricts).map((color, i) => (
               <Flex direction="column" align="center" key={i}>
-                <RadioGroupItem key={i} style={{backgroundColor: color}} value={color}>
+                <RadioGroupItem
+                  key={i}
+                  style={{backgroundColor: color, verticalAlign: 'top'}}
+                  value={String(i)}
+                  onContextMenu={e => {
+                    e.preventDefault();
+                    setColorSelectIndex(i);
+                  }}
+                >
                   <RadioGroupIndicator />
+                  <div
+                    style={{
+                      display: i === colorSelectIndex ? 'block' : 'none',
+                      position: 'absolute',
+                      left: 10,
+                      top: 2,
+                    }}
+                  >
+                    <TwitterPicker
```

Comment:
H: I think this is a bit tricky being a right click option in the paint menu. Customizing colors is an advanced option so I'd suggest the following workflow:

1. "Customize colors" as an option in the top left hamburger menu
2. Open a modal with the current colors in the plan
3. On clicking color, the TwitterPicker opens up inside the modal just below the colors (maybe defaults to customizing zone 1?)